src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let totalSum = 0;
+
+  const makeAdder = (number) => {
+    if (number === undefined) {
+      const result = totalSum;
+
+      totalSum = 0;
+
+      return result;
+    } else {
+      totalSum += number;
+
+      return makeAdder;
```

Comment:
Consider adding a comment above the `makeAdder` function to explain its behavior for future reference. While the code is clear, a brief explanation can help others understand the purpose of the function and how it works with the closure to accumulate values.